tests/utils/test_python_manager.py

```diff
@@ -85,39 +94,51 @@ def test_get_preferred_python_use_poetry_python_disabled_fallback(
 
 def test_fallback_on_detect_active_python(mocker: MockerFixture) -> None:
     m = mocker.patch(
-        "subprocess.check_output",
-        side_effect=subprocess.CalledProcessError(1, "some command"),
+        "findpython.find",
+        return_value=None,
     )
-
-    active_python = Python._detect_active_python(NullIO())
-
+    active_python = Python.get_active_python()
     assert active_python is None
     assert m.call_count == 1
 
 
 @pytest.mark.skipif(sys.platform != "win32", reason="Windows only")
-def test_detect_active_python_with_bat(tmp_path: Path) -> None:
+def test_detect_active_python_with_bat(
+    tmp_path: Path, without_mocked_findpython: None
+) -> None:
     """On Windows pyenv uses batch files for python management."""
     python_wrapper = tmp_path / "python.bat"
-    wrapped_python = Path(r"C:\SpecialPython\python.exe")
+
     encoding = "locale" if sys.version_info >= (3, 10) else None
     with python_wrapper.open("w", encoding=encoding) as f:
-        f.write(f"@echo {wrapped_python}")
+        f.write(
+            textwrap.dedent(f"""
+            @echo off
+            SET PYTHON_EXE="{sys.executable}"
+            %PYTHON_EXE% %*
+        """)
+        )
     os.environ["PATH"] = str(python_wrapper.parent) + os.pathsep + os.environ["PATH"]
 
-    active_python = Python._detect_active_python(NullIO())
+    python = Python.get_active_python()
+    assert python is not None
 
-    assert active_python == wrapped_python
+    # TODO: Asses if Poetry needs to discover real path in these cases as
+    # this is not a symlink and won't be handled by findpython
+    assert python.executable.as_posix() == python_wrapper.as_posix()
```

Comment:
@python-poetry/core as things stand findpython returns an object like this:

```
<PythonVersion executable=WindowsPath('C:/Users/.../python.bat'), version=<Version('3.12.2')>, architecture='64bit', implementation='CPython', major=3, minor=12, patch=2>
```

It correctly detects the version, but does not understand that this is a wrapper script. How do we want to handle this scenario?

I am guessing this will cause issues for folks that use a Python version installation manager that use `python.bat` to always have the same invocation while switching versions. But this also seems to be the case for cases where `python` is a symlink. 

For the latter we can use `findpython.PythonVersion.real_path` to reslove the path or simply `python.executable.resolve()`.